doc/cephfs/fs-volumes.rst

```diff
@@ -766,6 +766,377 @@ Will enable distributed subtree partitioning policy for the "csi" subvolume
+
+* **include** one or more subvolumes - quiesce set members
+* **exclude** one or more members
+* **cancel** the set, asynchronously aborting the pause on all its current members
+* **release** the set, requesting the end of the pause from all members and expecting an ack from all clients
+* **query** the current state of a set by id or all active sets or all known sets
+* **cancel all** active sets in case an immediate resume of IO is required.
+
+The operations listed above are non-blocking: they perform the intended modification if it's applicable
+and get back with an up to date version of the target set, whether the operation was successful or not. 
+The set may change states as a result of the modification, and the version that's returned in the response 
+is guaranteed to be in a state consistent with this and potentialy other successful operations from 
+the same control loop batch.
+
+Some set states are `awaitable`. We will discuss those below, but for now it's important to mention that
+any of the commands above can be amended with an **await** modifier, which will cause them to block
+on the set after applying their intended modification, as long as the resulting set state is `awaitable`.
+Such a command will block until the set reaches the awaited state, gets modified by another command,
+or transitions into another state. The reason for the unblock will be clear by the result code, while
+the contents of the response will always be the most recent set state.
+
+.. image:: quiesce-set-states.svg
+
+`Awaitable` states on the diagram are marked with ``(a)`` or ``(A)``. Blocking versions of the operations
+will pend while the set is in an ``(a)`` state and will complete with success if it reaches an ``(A)`` state.
+If the set is already at an ``(A)`` state, the operation completes immediately with a success. 
+
+Most of the operations require a set-id. The exceptions are:
+
+* creation of a new set without specifying a set id,
+* query of active or all known sets, and
+* the cancel all
+
+Creating a new set is achieved by including member(s) via the `include` or `reset` commands.
+It's possible to specify a set id, and if it's a new id then the set will be created
+with the specified member(s) in the `QUIESCING` state. When no set id is specified while including
+or resetting members, then a new set with a unique set id is created. The set id will be known
+to the caller by inspecting the output
+
+.. prompt:: bash $ auto
+
+  $ ceph fs subvolume quiesce a sub1 --set-id=unique-id
+  {
+      "epoch": 3,
+      "db_version": 1,
+      "sets": {
+          "unique-id": {
+              "db_version": 1,
+              "age_ref": 0.0,
+              "state": {
+                  "name": "TIMEDOUT",
+                  "age": 0.0
+              },
+              "timeout": 0.0,
+              "expiration": 0.0,
+              "members": {
+                  "file:/volumes/_nogroup/sub1/b1fcce76-3418-42dd-aa76-f9076d047dd3": {
```

Comment:
What does the `file:` prefix indicate? Can it ever be anything else?

Reply:
It can also be `inode:nnnnnnn`, but as of the current implementation we only work with `file:` roots.
The quiesce db will validate and sanitize the roots that are sent at the initial phase of request processing. If a root doesn't have a scheme then it's assumed to be an absolute path and is therefore converted to the `file:` format.

Reply:
A thorough explanation! thanks!